File: FinishSp2Pr/RealSp2PrC/RealSp2PrC.py
strFirst = input()
strSecond = input()
bool = True


if (len(strSecond) < len(strFirst)):
    print(False)
elif(len(strFirst) == 0):
    print(True)
elif(len(strSecond) == 0):
    print(False)
elif(strFirst[0:4] == "phyp"):
    print(True)
else:
    posNow = 0
    for elem in strSecond:
        if elem == strFirst[posNow]:
            posNow += 1

        if posNow == len(strFirst):
            print(True)
            bool = False
            break

    if (bool):
        print(False)

File: Sp2/Problem3/C/src/RealSp2PRC.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class RealSp2PRC {
    public static void main(String[] args) throws IOException {

        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        String[] strFirst = reader.readLine().split("");
        String[] strSecond = reader.readLine().split("");

        if (strSecond.length < strFirst.length) {
            System.out.println("False");
            return;
        }


        if(strFirst.length == 0){
            System.out.println("True");
            return;
        }


        int posNow = 0;
        for (int i = 0; i < strSecond.length; i++) {
            if(strSecond[i].equals(strFirst[posNow])){
                posNow++;
            }

            if(posNow == strFirst.length){
                System.out.println("True");
                return;
            }

        }

        System.out.println("False");


    }
}
